Directory.Build.props

```diff
@@ -25,6 +25,8 @@
     <RepositoryUrl>https://github.com/ClosedXML/ClosedXML</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
 
+    <!-- Sourcelink: https://github.com/dotnet/sourcelink/ -->
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
```

Comment:
I am not very familiar with making nuget packages. Do we need this for some reason? I looked an it seems that if we set RepositoryUrl (two lines above), it is redundant.

[PublishRepositoryUrl vs RepositoryUrl settings](https://github.com/dotnet/sourcelink/issues/659#issuecomment-711288537)

Reply:
At the moment we build a .nupkg and a .snupkg file. I have been uploading only the .nupkg file to NuGet thus far, with the idea of switching soon. I meant to do it for v0.96, but at that time I thought I could upload both, which is false. So the `RepositoryUrl` would affect the .nupkg build and the `PublishRepositoryUrl` would override that in the .snupkg build (not 100% sure, but I think that's true). I think this is minor enough to review later.